src/Middleware/HealthChecks/src/HealthCheckOptions.cs

```diff
@@ -24,16 +25,44 @@ public class HealthCheckOptions
         /// </remarks>
         public Func<HealthCheckRegistration, bool> Predicate { get; set; }
 
+        private IDictionary<HealthStatus, int> _resultStatusCodes = DefaultStatusCodesMapping;
```

Comment:
Unfortunately this isn't safe to do because it means the dictionary is shared across all instances of `HealthCheckOptions` within the same process. See the test failures here: https://github.com/aspnet/AspNetCore/pull/4796/checks?check_run_id=40915811

This will have to make a copy when you assign `DefaultStatusCodesMapping` to anything else.

Reply:
Fixed it by declaring the static `DefaultStatusCodesMapping` instance as `IReadOnlyDictionary` to force a copy when used.